packages/bundle-size-checker/src/fetchSnapshot.js

```diff
@@ -34,3 +34,61 @@ export async function fetchSnapshot(repo, sha) {
 
   throw new Error(`Failed to fetch snapshot`, { cause: lastError });
 }
+
+/**
+ * Gets parent commits for a given commit SHA using GitHub API
+ * @param {string} repo - Repository name (e.g., 'mui/material-ui')
+ * @param {string} commit - The commit SHA to start from
+ * @param {number} depth - How many commits to retrieve (including the starting commit)
+ * @returns {Promise<string[]>} Array of commit SHAs in chronological order (excluding the starting commit)
+ */
+async function getParentCommits(repo, commit, depth = 4) {
+  try {
+    const response = await fetch(
+      `https://api.github.com/repos/${repo}/commits?sha=${commit}&per_page=${depth}`,
+    );
+    if (!response.ok) {
+      throw new Error(`GitHub API request failed: ${response.status}`);
+    }
+
+    /** @type {{ sha: string }[]} */
+    const commits = await response.json();
+    // Skip the first commit (which is the starting commit) and return the rest
+    return commits.slice(1).map((commitDetails) => commitDetails.sha);
+  } catch (/** @type {any} */ error) {
+    console.warn(`Failed to get parent commits for ${commit}: ${error.message}`);
+    return [];
+  }
+}
+
+/**
+ * Attempts to fetch a snapshot with fallback to parent commits
+ * @param {string} repo - Repository name
+ * @param {string} commit - The commit SHA to start from
+ * @param {number} [fallbackDepth=3] - How many parent commits to try as fallback
+ * @returns {Promise<{snapshot: import('./sizeDiff').SizeSnapshot | null, actualCommit: string | null}>}
+ */
+export async function fetchSnapshotWithFallback(repo, commit, fallbackDepth = 3) {
```

Comment:
This reminds me of the Argos CI logic I wrote in the past, we had to walk the git commit tree to find a usable snapshot.

Reply:
That's also possible, but wouldn't work on the frontend